### DISCUSSION TITLE
How to change response status in OpenAPI spec?

### DISCUSSION THREAD

**Discussion by McMerph** [RobinTail/express-zod-api#824] [Q&A] [answered]
I'm using a custom EndpointsFactory that responds with HTTP status code 201. It's used for POST methods.
How can I change the response status from the default 200 to the actual 201 in the OpenAPI spec?

---
Actually I found the way:
```
...
const openApiSpec = new OpenAPI({...});

const { responses } = openApiSpec.rootDoc.paths["/api/v1/deploys"].post;
responses["201"] = responses["200"];
delete responses["200"];
...
```
But it looks ugly and error prone.
Does it make sense to add an additional field to the EndpointsFactory and pass the success HTTP status code to that field? I can help with the implementation of this feature, if such a feature is needed.

**Comment by RobinTail** [comment]
Oh, I'm sorry, @McMerph .
Currently, the documentation generator has no way to figure out which statuses your custom `ResultHandler` operates, so it uses `200` and `400` just for representation of a positive and a negative response. It is not precise in this place.
Perhaps, by equipping `ResultHandler` interface with a public method, like `getPositiveStatusCodes()`, we could achieve this.

**Reply by McMerph** [reply]
I don't understand why you want to name a new method like `getPositiveStatusCodes()`.
Why multiple status codes and not just one?

**Reply by McMerph** [reply]
I made a PR #841 which solves this problem.
But it has a public method `getPositiveStatusCode()` (singular) rather than `getPositiveStatusCodes()` (plural).

**Comment by RobinTail** [comment, marked as answer]
The feature released in v8.11.0 🚀 , @McMerph
